docs/adr/adr-006-fraud-service.md

```diff
@@ -0,0 +1,168 @@
+   required uint64 Height = 2;
+   repeated Share Shares = 3;
+   uint8 Position = 4;
+   bool isRow = 5;
+}
+```
+Data serialization/deserialization will be performed with `protobuf.Marshal`/`protobuf.Unmarshal`.
+
+`das.Daser` imports a data structure that implements `proof.FraudNotifier` interface that uses libp2p.pubsub under the hood:
+
+```go
+// FraudNotifier is a generic interface that sends a different kinds of fraud proofs to all subscribed on particular topic nodes
+type FraudNotifier interface {
+   // Broadcast takes a Fraud proof data stucture that implements standart BinaryMarshal interface and sends data to light nodes using libp2p pub-sub under the hood.
+   Notify(ctx context.Context, p Proof)  
+}
+```
+
+```go
+// FraudProofType is a enum type that represents particular fraud proof
+type FraudProofType string
+
+// Proof is a generic interface that will be used for all types of fraud proofs
+type Proof interface {
+   Type() FraudProofType
+   Height() (uint64, error)
+   Validate(*header.ExtendedHeader) error
+
+   encoding.BinaryMarshaller
+   encoding.BinaryUnmarshaller
+}
+```
+
+From the other side, light nodes have the ability to subscribe to a particular fraud proof update and verify received data:
+
+```go
+// Subscriber encompasses the behavior necessary to
+// subscribe/unsubscribe from new FraudProofs events from the
+// network.
+type Subscriber interface {
+   // Subscribe allows to subscribe on pub sub topic by it's type
+   Subscribe(ctx context.Context, proofType FraudProofType) (Subscription, error)
+}
+```
+
+```go
+// Subscription provides only valid proofs, because the validation is done under the hood of pub-sub
+type Subscription interface {
+   Proof() (Proof, error)
+}
+
+type FraudSub struct {
+   pubsub *pubsub.PubSub 
+}
+
+func NewFraudSub(p *pubsub.PubSub)(Subscription, error){}
+func(s *FraudSub) Proof() (Proof, error){}
+```
+
+```go
+// NOTE: re-think how FraudService should be designed(and constructed) for full nodes and for light nodes
+type FraudService struct {
+   badEncodingSub Subscription
+   fraudNotifier FraudNotifier
+}
+
+func(f *FraudService) Subscribe(ctx context.Context, proofType FraudProofType) (Subscription, error){}
+
+func(f *FraudService) Notify(ctx context.Context, p Proof){}
+```
+### BEFP verification
+Once a light node receives a `BadEncoding` fraud proof, it should:
+* verify that merkle proofs corresponds to particular shares(if merkle proof does not correspond to a share, than this BEFP is not valid)
+* using `BadEncoding.Shares` light node should re-construct full row or col, compute it's merkle root as in [rsmt2d](https://github.com/celestiaorg/rsmt2d/blob/master/extendeddatacrossword.go#L410) and compare it with merkle root that could be retreived from `dah.DataAvailabilityHeader`(if merkle roots do not match, then this BEFP is not valid)
+
+Light node should stop `DAS`, `Syncer` and `SubmitTx` services, in case if BEFP is valid.
+## Status
```

Comment:
```suggestion
### Bridge node behaviour
Bridge nodes will behave as light nodes do by subscribing to BEFP fraud sub and listening for BEFPs. If a BEFP is received, it will similarly shut down all dependent services, including broadcasting new `ExtendedHeader`s to the network.

## Status
```